src/payment/strategies/googlepay/googlepay-adyenv2-payment-processor.ts

```diff
@@ -0,0 +1,95 @@
+import { some } from 'lodash';
+
+import { Payment, PaymentActionCreator, PaymentInitializeOptions } from '../..';
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { MissingDataError, MissingDataErrorType, NotInitializedError, NotInitializedErrorType, RequestError } from '../../../common/error/errors';
+import { PaymentMethodCancelledError } from '../../errors';
+import { AdyenAction, AdyenAdditionalAction, AdyenAdditionalActionState, AdyenClient, AdyenError, AdyenV2ScriptLoader } from '../adyenv2';
+
+export default class GooglePayAdyenV2PaymentProcessor {
+    private _adyenClient?: AdyenClient;
+
+    constructor(private _store: CheckoutStore, private _paymentActionCreator: PaymentActionCreator, private _scriptLoader: AdyenV2ScriptLoader) {}
+
+    async initialize(options: PaymentInitializeOptions) {
+        const state = this._store.getState();
+        const paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(options.methodId);
+        const storeConfig = state.config.getStoreConfig();
+
+        if (!storeConfig) {
+            throw new MissingDataError(MissingDataErrorType.MissingCheckoutConfig);
+        }
+
+        const clientSideAuthentication = {
+            key: '',
+            value: '',
+        };
+
+        if (paymentMethod.initializationData.originKey) {
+            clientSideAuthentication.key = 'originKey';
+            clientSideAuthentication.value = paymentMethod.initializationData.originKey;
+        } else {
+            clientSideAuthentication.key = 'clientKey';
+            clientSideAuthentication.value = paymentMethod.initializationData.clientKey;
+        }
+
+        this._adyenClient = await this._scriptLoader.load({
+            environment:  paymentMethod.config.testMode ? 'TEST' : ' PRODUCTION',
+            locale: storeConfig.storeProfile.storeLanguage,
+            [clientSideAuthentication.key]: clientSideAuthentication.value,
+            paymentMethodsResponse: paymentMethod.initializationData.paymentMethodsResponse,
+        });
+    }
+
+    async _processAdditionalAction(error: unknown): Promise<InternalCheckoutSelectors> {
```

Comment:
this method seems to be public, so should probably not be prefixed with underscore